aspnetcore/signalr/hubs.md

```diff
@@ -0,0 +1,88 @@
+
+## Configure SignalR hubs
+
+The SignalR middleware requires some services, which are configured by calling `services.AddSignalR`.
+
+[!code-csharp[Configure service](hubs/sample/startup.cs?range=35)]
+
+When adding SignalR functionality to an ASP.NET Core app, setup SignalR routes by calling `app.UseSignalR` in the `Startup.Configure` method.
+
+[!code-csharp[Configure routes to hubs](hubs/sample/startup.cs?range=55-58)]
+
+## Create and use hubs
+
+Create a hub by declaring a class that inherits from `Hub`, and add public methods to it. Clients can call methods that are defined as `public`.
+
+[!code-csharp[Create and use hubs](hubs/sample/chathub.cs?range=10-13)]
+
+You can specify a return type and parameters, including complex types and arrays, as you would in any C# method. SignalR handles the serialization and deserialization of complex objects and arrays in your parameters and return values.
+
+## The Clients object
+
+Each instance of the `Hub` class has a property named `Clients` that contains the following members for communication between server and client:
+
+| Property | Description |
+| ------ | ----------- |
+| `All` | Calls a method on all connected clients |
+| `Caller` | Calls a method on the client that invoked the hub method |
+| `Others` | Calls a method on all connected clients except the client that invoked the method |
+
+Additionally, the `Hub` class contains the following methods:
+
+| Method | Description |
+| ------ | ----------- |
+| `AllExcept` | Calls a method on all connected clients except for the specified connections |
+| `Client` | Calls a method on a specific connected client |
+| `Clients` | Calls a method on specific connected clients |
+| `Group` | Sends a message to all connections in the specified group  |
+| `GroupExcept` | Sends a message to all connections in the specified group, except the specified connections |
+| `Groups` | Sends a message to multiple groups of connections  |
+| `OthersInGroup` | Sends a message to a group of connections, excluding the client that invoked the hub method  |
+| `User` | Sends a message to all connections associated with a specific user |
+| `Users` | Sends a message to all connections associated with the specified users |
+
+Each property or method in the preceding tables returns an object with a `SendAsync` method. The `SendAsync` method allows you to supply the name and parameters of the client method to call.
+
+## Send messages to clients
+
+To make calls to specific clients, use the `Clients.Client` or `Clients.Clients`. In the following example, the `SendMessageToCaller` method demonstrates sending a message to the connection that invoked the hub method. The `SendMessageToGroups` method sends a message to the groups stored in a `List` named `groups`.
```

Comment:
We talk about `Client`/`Clients` in the first sentence, then cover different methods later in the paragraph. How about we just generalize the first sentence:

To make calls to specific clients, use the properties of the `Clients` object. In the following...